pkg/resource/model_package/resource.go

```diff
@@ -92,10 +92,15 @@ func (r *resource) SetStatus(desired acktypes.AWSResource) {
 // SetIdentifiers sets the Spec or Status field that is referenced as the unique
 // resource identifier
 func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
-	if identifier.NameOrID == "" {
-		return ackerrors.MissingNameIdentifier
+	if r.ko.Status.ACKResourceMetadata == nil {
+		r.ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
+	}
+	r.ko.Status.ACKResourceMetadata.ARN = identifier.ARN
+
+	f0, f0ok := identifier.AdditionalKeys["modelPackageName"]
+	if f0ok {
+		r.ko.Spec.ModelPackageName = &f0
```

Comment:
shouldn't this be removed?

we just want to adopt using ARN for all cases right?

also, please add an integ test for this since its different from other resources

Reply:
Atm looks like code-gen will add those lines of code regardless. I have set primary identifier to ARN. 
Added integ test for adopting versioned model package

Reply:
ok for now, let's come back to this